test/unit/alphabet/structure/structure_test_template.hpp

```diff
@@ -0,0 +1,177 @@
+            "S"_wuss51.front()
+        };
+    }
+    else if constexpr (std::is_same_v<TypeParam, dssp9>)
+    {
+        cmp =
+        {
+            t::X, t::X, t::X,
+            t::X, t::X, t::X, t::X, t::X, t::X, t::X, t::X, t::X, t::X, t::X, t::X,
+            t::H, t::B, t::E, t::G, t::I, t::T, t::S
+        };
+    }
+
+    for (auto [ ch, cm ] : ranges::view::zip(input, cmp))
+        EXPECT_EQ((assign_char(TypeParam{}, ch)), cm);
+}
+
+// to_char functions
+TYPED_TEST_P(structure, global_to_char)
+{
+    if constexpr (std::is_same_v<TypeParam, dot_bracket3>)
+    {
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED), '.');
+        EXPECT_EQ(to_char(TypeParam::PAIR_OPEN), '(');
+        EXPECT_EQ(to_char(TypeParam::PAIR_CLOSE), ')');
+        EXPECT_EQ(to_char(TypeParam::UNKNOWN), '.');
+    }
+    else if constexpr (std::is_same_v<TypeParam, wuss51>)
+    {
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED), '.');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED1), ':');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED2), ',');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED3), '-');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED4), '_');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED5), '~');
+        EXPECT_EQ(to_char(TypeParam::UNPAIRED6), ';');
+        EXPECT_EQ(to_char(TypeParam::PAIR_OPEN), '<');
+        EXPECT_EQ(to_char(TypeParam::PAIR_CLOSE), '>');
+        EXPECT_EQ(to_char(TypeParam::PAIR_OPEN1), '(');
+        EXPECT_EQ(to_char(TypeParam::PAIR_CLOSE1), ')');
+        EXPECT_EQ(to_char(TypeParam::PAIR_OPEN2), '[');
+        EXPECT_EQ(to_char(TypeParam::PAIR_CLOSE2), ']');
+        EXPECT_EQ(to_char(TypeParam::PAIR_OPEN3), '{');
+        EXPECT_EQ(to_char(TypeParam::PAIR_CLOSE3), '}');
+    }
+    else if constexpr (std::is_same_v<TypeParam, dssp9>)
+    {
+        EXPECT_EQ(to_char(TypeParam::H), 'H');
+        EXPECT_EQ(to_char(TypeParam::B), 'B');
+        EXPECT_EQ(to_char(TypeParam::E), 'E');
+        EXPECT_EQ(to_char(TypeParam::G), 'G');
+        EXPECT_EQ(to_char(TypeParam::I), 'I');
+        EXPECT_EQ(to_char(TypeParam::T), 'T');
+        EXPECT_EQ(to_char(TypeParam::S), 'S');
+        EXPECT_EQ(to_char(TypeParam::C), 'C');
+        EXPECT_EQ(to_char(TypeParam::X), 'X');
+    }
+}
+
+// ------------------------------------------------------------------
+// concepts
+// ------------------------------------------------------------------
+
+TYPED_TEST_P(structure, concept_check)
+{
+    EXPECT_TRUE(alphabet_concept<TypeParam>);
+}
+
+TYPED_TEST_P(structure, global_rna_structure_concept)
+{
+    if constexpr (!std::is_same_v<TypeParam, dssp9>)
+    {
+        EXPECT_TRUE(rna_structure_concept<TypeParam>);
+        EXPECT_NE(max_pseudoknot_depth_v<TypeParam>, 0);
+    }
+    else
+    {
+        EXPECT_FALSE(rna_structure_concept<TypeParam>);
+    }
+
+    // additional wuss test
+    EXPECT_TRUE(rna_structure_concept<wuss<>>);  // also wuss51
+    EXPECT_TRUE(rna_structure_concept<wuss<67>>);
+}
```

Comment:
The stuff in this file is literally just three different tests. It was ok to group these if everything is in one file anyway, but if you split this, please also split these tests into the individual files.